zrml/prediction-markets/src/migrations.rs

```diff
@@ -0,0 +1,138 @@
+                T::MarketId,
+                Option<PoolId>,
+                Blake2_128Concat,
+            >(PM, b"MarketToSwapPool")
+            .filter_map(|(k, v)| Some((k, v?)))
+            {
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+                T::MarketCommons::insert_market_pool(k, v);
+            }
+        }
+
+        weight
+    }
+
+    #[cfg(test)]
+    mod test {
+        use super::*;
+        use crate::mock::{Court, ExtBuilder, MarketCommons, PredictionMarkets};
+        use frame_support::{traits::OnRuntimeUpgrade, Hashable};
+        use zeitgeist_primitives::types::{
+            Market, MarketCreation, MarketEnd, MarketStatus, MarketType,
+        };
+
+        #[test]
+        fn migration_works() {
+            const DEFAULT_MARKET: Market<u128, u64> = Market {
+                creation: MarketCreation::Permissionless,
+                creator_fee: 0,
+                creator: 0,
+                end: MarketEnd::Block(0),
+                market_type: MarketType::Categorical(0),
+                metadata: vec![],
+                oracle: 0,
+                report: None,
+                resolved_outcome: None,
+                status: MarketStatus::Closed,
+            };
+            ExtBuilder::default().build().execute_with(|| {
+                assert!(Court::market_ids_per_dispute_block(&0).is_err());
+                assert!(Court::market_ids_per_report_block(&0).is_err());
+
+                assert!(MarketCommons::market(&0).is_err());
+                assert!(MarketCommons::market_pool(&0).is_err());
+
+                frame_support::migration::put_storage_value(
+                    PM,
+                    b"MarketIdsPerDisputeBlock",
+                    &0u128.blake2_128_concat(),
+                    vec![1u128],
+                );
+                frame_support::migration::put_storage_value(
+                    PM,
+                    b"MarketIdsPerReportBlock",
+                    &0u128.blake2_128_concat(),
+                    vec![1u128],
+                );
+
+                frame_support::migration::put_storage_value(
+                    PM,
+                    b"Markets",
+                    &0u128.blake2_128_concat(),
+                    Some(DEFAULT_MARKET),
+                );
+                frame_support::migration::put_storage_value(
+                    PM,
+                    b"MarketToSwapPool",
+                    &0u128.blake2_128_concat(),
+                    Some(1u128),
+                );
+
+                PredictionMarkets::on_runtime_upgrade();
+
+                assert_eq!(Court::market_ids_per_dispute_block(&0).unwrap(), vec![1]);
+                assert_eq!(Court::market_ids_per_report_block(&0).unwrap(), vec![1]);
+
+                assert_eq!(MarketCommons::market(&0).unwrap(), DEFAULT_MARKET);
+                assert_eq!(MarketCommons::market_pool(&0).unwrap(), 1);
+            })
```

Comment:
Some additional tests that ensure, that the prediction market pallet does not have those values in storage anymore.